csharp/extractor/Semmle.Util/FileUtils.cs

```diff
@@ -144,20 +144,40 @@ public static string NestPaths(ILogger logger, string? outerpath, string innerpa
             return nested;
         }
 
+        private static string? tempFolderPath = null;
```

Comment:
Since we are introducing state, would it make sense to re-factor this into a class that needs to be instantiated (just the GetTemp work dir parts)?

Reply:
I applied your `private static Lazy<string> tempFolderPath` suggestion from below. I think there's no need for another class that way.